rtlib/src/SepDisegni.cpp

```diff
@@ -0,0 +1,275 @@
+    Mat dst = Mat::zeros(markers.size(), CV_8UC3);
+    // Fill labeled objects with random colors
+    for (int i = 0; i < markers.rows; i++) {
+        for (int j = 0; j < markers.cols; j++) {
+            int index = markers.at<int>(i, j);
+            if (index > 0 && index <= static_cast<int>(contours.size())) {
+                dst.at<Vec3b>(i, j) = colors[index - 1];
+            }
+        }
+    }
+    // Visualize the final image
+    namedWindow("Final Result", 0);
+    imshow("Final Result", dst);
+    waitKey(0);
+
+    IntensityImage_ = dst;
+
+
+    return markers;
+}
+
+
+
+
+
+
+vector<Mat> SepDisegni::split_merge(Mat watershededImage) {
+
+
+    /* std::map<int32_t, int32_t> labelCounts;
+    for( int y = 0; y < watershededImage.rows; y++) {
+        for( int x = 0; x < watershededImage.cols; x++) {
+            auto label = watershededImage.at<int32_t>(y,x);
+            if(labelCounts.count(label) > 0) {
+                labelCounts[label] += 1;
+            } else {
+                labelCounts[label] = 1;
+            }
+        }
+    } */
+
+    //Finding Boundary Boxes
+    static const int INT_MINI = std::numeric_limits<int>::min();
+    static const int INT_MAXI = std::numeric_limits<int>::max();
+
+    struct MinMax {
+        cv::Point min{INT_MAXI, INT_MAXI};
+        cv::Point max{INT_MINI, INT_MINI};
+    };
+
+    //Finding Maximum and Minimum of a subregion
+    std::map<int32_t, MinMax> extremeFinder;
+    for( int y = 0; y < watershededImage.rows; y++) {
+        for( int x = 0; x < watershededImage.cols; x++) {
+
+            //get label
+            auto label = watershededImage.at<int32_t>(y,x);
+
+            //Labels that are skipped because they aren't a part of the subregion
+            if(label == -1 || label == 255) { continue; }
+
+            //Make a new minmax if we don't have one for this label
+            if(extremeFinder.count(label) == 0) {
+                extremeFinder[label] = MinMax();
+                continue;
+            }
+
+            //This finds the max (top-left point) and min (bottom-right point) for each subregion
+            if (x < extremeFinder[label].min.x ) { extremeFinder[label].min.x = x; };
+            if (y < extremeFinder[label].min.y ) { extremeFinder[label].min.y = y; };
+            if (x > extremeFinder[label].max.x ) { extremeFinder[label].max.x = x; };
+            if (y > extremeFinder[label].max.y ) { extremeFinder[label].max.y = y; };
+
+
+
+
+        }
+    }
+
+    //Region of Interest taking Minimum and Maximum points
+    /*int height = ((extremeFinder[1].max.y) - (extremeFinder[1].min.y));
+    int width = ((extremeFinder[1].max.x) - (extremeFinder[1].min.x));
+    Rect region_of_interest = Rect(extremeFinder[1].min.x,extremeFinder[1].min.y, width, height);
+    Mat image_roi = image_(region_of_interest).clone(); */
+
+    vector<Mat> regionsOfInterest;
+    for(auto i = extremeFinder.begin(); i != extremeFinder.end(); i++){
```

Comment:
Use range-based for loop. Since we're only referencing the minmaxes we found, instead of modifying them, it can be a const loop:
```c++
for(const auto& e : extremeFinder) {
    int height = e.second.max.y - e.second.min.y;
    int width = e.second.max.x - e.second.min.x;
    ....
}
```
Note: Range-based loops pass objects/references to objects, not pointers, so you don't use the `->` operator.